hw07.c

```diff
@@ -0,0 +1,246 @@
+//zde musem volat fci s fcnim ukazatel na pole a pdle first/,... hodnot se zavola rovnou dana fce,,nepamatujeu se uz inicializaci atd
+//POLE FUNKCNICHU UKAZATELU
+    void (*func_array[])(TGAImage*, int, int, RGBA*) = {
+        draw_zero, draw_one, /*draw_two, draw_three, draw_foure, draw_five, draw_six, draw seven, draw_eight, draw_nine*/
+    };
+//jen budu omezovat nebo pricit konec zacatek k ouradnicim(prevod do 1D to pak prepocita na ta mista v celem poli, ktere je uz predpripraveno)
+    //draw_one(self,0,0,&fg);
+    /*void (*fptr)(TGAImage*, int, int, RGBA*);
+    fptr = &draw_one;*/
+
+   /* func_array[firstH](self, 0, 0, &fg);
+    func_array[secondH](self, 184, 0, &fg);
+    func_array[firstMin](self, 0, 224, &fg);
+    func_array[secondMin](self, 184, 224, &fg);*/
+
+    int writeCount = fwrite( &self->header, sizeof (TGAheader),1,file); //velikost prvku je header celej a je jen jeden
+    assert(writeCount == 1);
+
+    writeCount = fwrite(&self->data, sizeof(RGBA), tga_width(self) * tga_height(self) * sizeof(RGBA),file);
+
+    fclose(file);
+}
+
+
+//POTREBUJU INIT BRIGHTNESS NENI TO pro cernobile nebo c oto kurva je
+TGAImage *tga_new(const int height, const int width){
+    TGAImage *tga = (TGAImage*)malloc(sizeof(TGAImage));// na heapu takze to neni jen loklani na stacku
+  //  ALLOCTEST( TGAImage );
+
+    memset(&tga->header, 0, sizeof(TGAheader)); //potrebuje adresu ukazatele na novy objekt tga, pac fce to chce
+//TADY CHYBI NEKERE HODNOTY
+    //tga->header == (*tga).header
+    tga->header.image_type = 2;
+    tga->header.depth = 32;
+    tga->header.descriptor |= 1 << 3;
+    tga->header.descriptor |= 1 << 5;
+
+
+    //zapis rozmeru na dva bytes, prvni vyssii jsou nuly, pak cislo 0-255 zapsane binarne v nizsim bytu
+    memcpy(&tga->header.width, &width, 2 );
+    memcpy(&tga->header.height, &height, 2 );
+
+    //ted alokuje pamet pro pixely podle = velikost struct pixel x vyska x sirka
+    tga->data = (RGBA*)malloc(sizeof(RGBA) * width * height); //data je typu RGBA*
+    //nastavime celou alokovanou  pamet pro data
+    memset(tga->data, 0, sizeof(RGBA) * width * height); //tga->data uz je ukazatel
+
+    return tga;
+
+}
+
+void tga_free(TGAImage **self){
+
+    free((*self)->data);
+    (*self)->data = NULL;
+    free(*self);
+    *self = NULL;
+
+}
+
+
+int main(int argc, char **argv){
+    int hours = 0;
+    int minutes = 0;
+
+    if (argc != 3)
+    {
+        return 1;
+    }
+    else
+    {
+        hours = strtol(argv[1],NULL,10);
+        minutes = strtoll(argv[2], NULL, 10);
+    }
+
+    //soubor otevru tady nebo muzu presunout do zapisovaci fce?
+
+
+    //vytvorim strukturu obrazku
+    TGAImage *self = tga_new(448,368);
+
+    //bytvorim struktury pixelu kt. priradim urcite hodnoty tak aby vznikly ruzne barvy pozadi a popredi
+    RGBA bg = {.red = 220, .green = 220, .blue = 20, .alpha = 255};
+    RGBA fg = {.red = 100, .green = 10, .blue = 10, .alpha = 255};
+
+    draw_bg(self, &bg);
```

Comment:
Opet, tu to nedava smysl, ma to byt ve `watch_draw_time`, nebo u alokace... A taky to muzes kompletne pouzit misto nulovani pameti skrze `memset`.

Reply:
![image](https://user-images.githubusercontent.com/68806256/146641387-b49efbb5-5271-4cf1-8c46-1173d302b4ad.png)
a k te halvicce, v debuggeru tohle to ukazuje v debuggeru, jakoze nerozumim moc jak se interpretuji ty dve hodoty heigth width zapsane na dva bajty, ale ta funcke co vraci height, width z headru vraci spravna cisla, tak furt nevim, co je tam tak blbe, ze nedokazu nic zapsat do toho obrazku kterej dyzky vypada takhle

Reply:
![image](https://user-images.githubusercontent.com/68806256/146641466-015fc7bc-455f-4a63-a74f-5ec575c4e8a1.png)

Reply:
ok tak sem to nasel ten problem
 writeCount = fwrite(&self->data, sizeof(RGBA), tga_width(self) * tga_height(self),file);
ma byt:
 writeCount = fwrite(self->data, sizeof(RGBA), tga_width(self) * tga_height(self),file);
predaval sem adresu na ukazatel misto primo ukazatel, kurva tohle me nici!!!
![image](https://user-images.githubusercontent.com/68806256/146643039-64a67f5b-9365-403e-b078-0becf348cc2f.png)

Reply:
JJ pozor na ukazatele.